e2e/components/DataTable/DataTable-test.avt.e2e.js

```diff
@@ -0,0 +1,373 @@
+          theme: 'white',
+        },
+      });
+      await expect(page).toHaveNoACViolations(
+        'components-datatable-batch-actions--default'
+      );
+    });
+  });
+
+  test.describe('dynamic', () => {
+    test('default has no accessibility-checker violations', async ({
+      page,
+    }) => {
+      await visitStory(page, {
+        component: 'DataTable',
+        id: 'components-datatable-dynamic--default',
+        globals: {
+          theme: 'white',
+        },
+      });
+      await expect(page).toHaveNoACViolations(
+        'components-datatable-dynamic--default'
+      );
+    });
+
+    test('default keyboard navigation sequence', async ({ page }) => {
+      await visitStory(page, {
+        component: 'DataTable',
+        id: 'components-datatable-dynamic--default',
+        globals: {
+          theme: 'white',
+        },
+      });
+
+      await expect(page).toHaveNoACViolations(
+        'components-datatable-dynamic--default'
+      );
+
+      // Start off by manually focusing the search input
+      await page.getByRole('searchbox').focus();
+      await expect(page.getByRole('searchbox')).toBeFocused();
+
+      // Navigate to the gear/settings button
+      await page.keyboard.press('Tab');
+      await expect(
+        page.getByRole('button', { name: 'Settings' })
+      ).toBeFocused();
+
+      // Navigate to the select all checkbox
+      await page.keyboard.press('Tab');
+      await expect(
+        page.getByRole('checkbox', { name: 'Select all rows' })
+      ).toBeFocused();
+
+      // Pressing enter on the select all checkbox shouldn't do anything
+      await page.keyboard.press('Enter');
+      await expect(
+        page.getByRole('checkbox', { name: 'Select all rows' })
+      ).not.toBeChecked();
+
+      // Pressing space should check the select all checkbox
+      await page.keyboard.press('Space');
+      await expect(
+        page.getByRole('checkbox', { name: 'Select all rows' })
+      ).toBeChecked();
+
+      // Every checkbox should be checked
+      for (const checkbox of await page.getByRole('checkbox').all()) {
+        await expect(checkbox).toBeChecked();
+      }
+
+      // Pressing space should uncheck the select all checkbox
+      await page.keyboard.press('Space');
+      await expect(
+        page.getByRole('checkbox', { name: 'Select all rows' })
+      ).not.toBeChecked();
+      // Every checkbox should no longer be checked
+      for (const checkbox of await page.getByRole('checkbox').all()) {
+        await expect(checkbox).not.toBeChecked();
+      }
+
+      // Navigate to the first expansion button
+      await page.keyboard.press('Tab');
+      await expect(
+        page.getByRole('button', { name: 'Expand current row' }).first()
+      ).toBeFocused();
+      // Expand the first row
+      await page.keyboard.press('Space');
+      await expect(
+        page.getByRole('heading', { name: 'Expandable row content' }).first()
+      ).toBeVisible();
```

Comment:
This test seems to pass regardless if the expandable row is toggled or not since the expandable row is always in the DOM. I'm guessing this is so we can animate the expansion, but I'm wondering if we should be toggling `aria-expanded` on `data-child-row` elements so that we can test this a bit better. 

Related: https://github.com/carbon-design-system/carbon/issues/14261

Reply:
Yeah I noticed this as well. I've actually got https://github.com/carbon-design-system/carbon/issues/14262 in my queue for this sprint, which I think will fix #14261 as well.

Reply:
Awesome 👍🏻